tests/Integration/PrestaShopBundle/Controller/Admin/Order/DeliveryControllerTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Integration\PrestaShopBundle\Controller\Admin\Order;
+
+use Symfony\Component\HttpFoundation\Response;
+use Tests\Integration\PrestaShopBundle\Test\WebTestCase;
+
+/**
+ * @group demo
+ */
+class DeliveryControllerTest extends WebTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+        $this->enableDemoMode();
+    }
+
+    public function testSlipAction()
+    {
+        $this->client->request(
+            'GET',
+            $this->router->generate(
+                'admin_order_delivery_slip'
+            )
+        );
+
+        $this->assertEquals(Response::HTTP_OK, $this->client->getResponse()->getStatusCode());
```

Comment:
improvement for another PR: add new specific assertions like assertResponseOk(Response $response)

I have the feeling that @tomlev have already done that somewhere...

Reply:
Yes but it doesn't return the reponse, we can't continue the test and we can't customize the route and (again) it's for API, will maybe in another PR :)
https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/PrestaShopBundle/Controller/Api/ApiTestCase.php#L229

Reply:
> improvement for another PR

> will maybe in another PR

<3 totally, the more we do quality, the more we want to increase the quality: hard to keep the focus on such exciting times \o/